msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MappedByteBufferInputStream.java

```diff
@@ -19,35 +18,41 @@
 import java.nio.MappedByteBuffer;
 
 class MappedByteBufferInputStream extends InputStream {
-	private MappedByteBuffer buf;
-
-	MappedByteBufferInputStream(MappedByteBuffer buf) {
-		this.buf = buf;
-	}
-
-	public synchronized int read() throws IOException {
-		if (!buf.hasRemaining()) {
-			return -1;
-		}
-		int readBuffer = buf.get();
-		return readBuffer;
-	}
-
-	public synchronized int read(byte[] bytes, int off, int len) throws IOException {
-		if (!buf.hasRemaining()) {
-			return -1;
-		}
-
-		len = Math.min(len, buf.remaining());
-		buf.get(bytes, off, len);
-		return len;
-	}
-
-	public synchronized int getCurrentPosition() {
-		return buf.position();
-	}
-
-	public synchronized void setPosition(int pos) {
-		buf = (MappedByteBuffer) buf.position(pos);
-	}
+  private MappedByteBuffer buf;
+
+  MappedByteBufferInputStream(MappedByteBuffer buf) {
+    this.buf = buf;
+  }
+
+  public synchronized int read() throws IOException {
+    if (!buf.hasRemaining()) {
+      return -1;
+    }
+    int readBuffer = buf.get();
+    return readBuffer;
+  }
+
+  public synchronized int read(byte[] bytes, int off, int len) throws IOException {
+    if (!buf.hasRemaining()) {
+      return -1;
+    }
+
+    len = Math.min(len, buf.remaining());
+    buf.get(bytes, off, len);
+    return len;
+  }
+
+  public synchronized InputStream getInputStream(int position, int length) {
+    MappedByteBuffer newBuf = (MappedByteBuffer) buf.position(position);
```

Comment:
You need to make a duplicate of the byte buffer before setting the position. Otherwise, if two threads are using the same MappedByteBuffer, it will mess up their input (the positions have to be independent).